mwscan/tests/test_mwscan.py

```diff
@@ -85,9 +85,15 @@ def test_find_targets_will_exclude_patterns(self):
         files = list(scan.find_targets(self.target_path, exclude_patterns=p))
         assert files == [], files
 
-    def test_that_symlinks_are_followed(self):
+    def test_that_symlinks_are_properly_followed(self):
         paths = scan.find_targets(self.target_path, follow_symlinks=True)
-        files = map(os.path.basename, paths)
+        files = list(map(os.path.basename, paths))
+
+
+        # py3 returns map whereas py2 returns list,
+        # map can only be iterated once
+        assert len(files) > 5, 'bogus files obj'
```

Comment:
Looks good. If I understand correct, it will stop traversing when the basename appears more than 5 times in the path?

Reply:
No, this is the unit test of scan_targets, which asserts that the fixture data contains at least 5 files. But I see that I wrote a very confusing comment :)

Reply:
The real recursive symlink fix is in https://github.com/gwillem/magento-malware-scanner/blob/fcda7b0cbe1ab12857ccc7f0f54e3716350ee7bb/mwscan/scan.py#L137, which maintains a set of already visited dirs (based on device+inode) and only add new ones to the target list.